flytekit/bin/entrypoint.py

```diff
@@ -222,98 +224,141 @@ def _handle_annotated_task(task_def: PythonTask, inputs: str, output_prefix: str
-                    # TODO: AWS batch array jobs.
-                    _flyte_random.seed_flyte_random(
-                        "{} {} {}".format(_random.random(), _datetime.datetime.utcnow(), job_index)
-                    )
+                # TODO: Perhaps remove.  This is a workaround to an issue we perceived with limited entropy in
+                # TODO: AWS batch array jobs.
+                _flyte_random.seed_flyte_random(
+                    "{} {} {}".format(_random.random(), _datetime.datetime.utcnow(), job_index)
+                )
 
-                    # If an ArrayTask is discoverable, the original job index may be different than the one specified in
-                    # the environment variable. Look up the correct input/outputs in the index lookup mapping file.
-                    job_index = _map_job_index_to_child_index(input_dir, inputs, job_index)
+                # If an ArrayTask is discoverable, the original job index may be different than the one specified in
+                # the environment variable. Look up the correct input/outputs in the index lookup mapping file.
+                job_index = _map_job_index_to_child_index(input_dir, inputs, job_index)
 
-                    inputs = _os.path.join(inputs, str(job_index), "inputs.pb")
-                    output_prefix = _os.path.join(output_prefix, str(job_index))
+                inputs = _os.path.join(inputs, str(job_index), "inputs.pb")
+                output_prefix = _os.path.join(output_prefix, str(job_index))
 
-                _data_proxy.Data.get_data(inputs, local_inputs_file)
-                input_proto = _utils.load_proto_from_file(_literals_pb2.LiteralMap, local_inputs_file)
+            _data_proxy.Data.get_data(inputs, local_inputs_file)
+            input_proto = _utils.load_proto_from_file(_literals_pb2.LiteralMap, local_inputs_file)
 
-                _engine_loader.get_engine().get_task(task_def).execute(
-                    _literal_models.LiteralMap.from_flyte_idl(input_proto),
-                    context={"output_prefix": output_prefix, "raw_output_data_prefix": raw_output_data_prefix},
-                )
+            _engine_loader.get_engine().get_task(task_def).execute(
+                _literal_models.LiteralMap.from_flyte_idl(input_proto),
+                context={"output_prefix": output_prefix, "raw_output_data_prefix": raw_output_data_prefix},
+            )
 
-            # New core style task
-            elif not test and isinstance(task_def, PythonTask):
-                _handle_annotated_task(task_def, inputs, output_prefix, raw_output_data_prefix)
+
+def _load_resolver(resolver_location: str) -> TaskResolverMixin:
+    # Load the actual resolver - this cannot be a nested thing, whatever kind of resolver it is, it has to be loadable
+    # directly from importlib
+    # TODO: Handle corner cases, like where the first part is [] maybe
+    resolver = resolver_location.split(".")
```

Comment:
can you add a sample of what the resolver_location/resolver looks like? hard to see the big picture with the indexing below